### PR TITLE
[project-pineapples < T0951-MG] Create query engine v2

### PR DESCRIPTION
**Pull request by jbajic**
[project-pineapples < T0951-MG] PR
- [ ] ~~Check, and update documentation if necessary~~
- [ ] ~~Update [changelog](https://docs.memgraph.com/memgraph/changelog)~~
- [x] Final commit message
- Create version v2 of the query engine.
- Adjust CMake and lisp files
- Connect query engine v2 with storage engine v3

### CODE REVIEW

File: src/query/v2/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,104 @@
+define_add_lcp(add_lcp_query lcp_query_2_cpp_files generated_lcp_query_2_files)
```
Comment (antaljanosbenjamin) [inline src/query/v2/CMakeLists.txt:1]: Can this be `lcp_query_v2`? Just to be consistent how the lcp related things are handled in storage v2/v3.

Comment (antaljanosbenjamin) [reply src/query/v2/CMakeLists.txt:1]: And I just noticed I also didn't update this properly in storage-v3! Note for myself.

File: src/query/v2/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    ${opencypher_frontend}/grammar/Cypher.g4)
+
+add_custom_target(generate_opencypher_parser_2
+    DEPENDS ${antlr_opencypher_generated_src} ${antlr_opencypher_generated_include})
+
+add_library(antlr_opencypher_parser_lib_2 STATIC ${antlr_opencypher_generated_src})
+add_dependencies(antlr_opencypher_parser_lib_2 generate_opencypher_parser_2)
+target_link_libraries(antlr_opencypher_parser_lib_2 antlr4)
+
+target_link_libraries(mg-query-v2 antlr_opencypher_parser_lib_2)
```
Comment (antaljanosbenjamin) [inline src/query/v2/CMakeLists.txt:104]: Also, can this be v2?

**Review by antaljanosbenjamin — CHANGES_REQUESTED** [review]
I messed up the review, but my comments can be read above. Sorry for this.

File: .github/workflows/release_centos8.yaml [hunk 3]
```diff
@@ -128,7 +128,7 @@ jobs:
 
           # Build debug binaries.
           cd build
-          cmake ..
+          cmake ..sas
```
[the remark this answers is not in the document]
Comment (jbajic) [reply .github/workflows/release_centos8.yaml:131]: Wow I did not notice this

**Review by antaljanosbenjamin — APPROVED** [review]
Because the code is just copied, we are merging this without eliminating all of the clang-tidy warnings.

**Pull request merged**